google/cloud/bigquery/model.py

```diff
@@ -272,140 +243,117 @@ def encryption_configuration(self):
         return prop
 
     @encryption_configuration.setter
-    def encryption_configuration(self, value):
+    def encryption_configuration(self, value: Optional[EncryptionConfiguration]):
         api_repr = value
         if value:
             api_repr = value.to_api_repr()
         self._properties["encryptionConfiguration"] = api_repr
 
     @classmethod
-    def from_api_repr(cls, resource: dict) -> "Model":
+    def from_api_repr(cls, resource: Dict[str, Any]) -> "Model":
         """Factory: construct a model resource given its API representation
 
         Args:
-            resource (Dict[str, object]):
+            resource:
                 Model resource representation from the API
 
         Returns:
-            google.cloud.bigquery.model.Model: Model parsed from ``resource``.
+            Model parsed from ``resource``.
         """
         this = cls(None)
-        # Keep a reference to the resource as a workaround to find unknown
-        # field values.
+
+        resource = copy.deepcopy(resource)
         this._properties = resource
 
         # Convert from millis-from-epoch to timestamp well-known type.
         # TODO: Remove this hack once CL 238585470 hits prod.
-        resource = copy.deepcopy(resource)
         for training_run in resource.get("trainingRuns", ()):
```

Comment:
Regardless of whether the original change to well-known types is in production (it should be as that change was submitted in Mar 15, 2019), I think we should remove this. The goal is not to transform the data for fields such as `trainingRuns` and just give the developer exactly what the API returns.